### PR TITLE
[FLINK-10880] Exclude JobManagerOptions#EXECUTION_FAILOVER_STRATEGY from documentation

### PR DESCRIPTION
**Pull request by tillrohrmann**
## What is the purpose of the change

This commit excludes the `JobManagerOptions#EXECUTION_FAILOVER_STRATEGY` from Flink's
configuration documentation.

## Brief change log

- Introduce `Documentation#ExcludeFromDocumentation`
- Exclude `JobManagerOptions#EXECUTION_FAILOVER_STRATEGY` from documentation

## Verifying this change

- Added `ConfigOptionsDocGeneratorTest#testConfigOptionExclusion`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java [hunk 1]
```diff
@@ -106,6 +106,7 @@
 	/**
 	 * This option specifies the failover strategy, i.e. how the job computation recovers from task failures.
 	 */
+	@Documentation.ExcludeFromDocumentation
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java]: Do we maybe want to include a reason for why this is hidden similar to deprecation notices?

File: flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java [hunk 2]
```diff
@@ -213,6 +213,12 @@ static void processConfigOptions(String rootDir, String module, String packageNa
 		}
 	}
 
+	private static boolean isValidConfigOption(Field field) {
```
Comment (zentol) [inline flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java]: I'm not fond of the naming as a deprecated options is still valid.

Maybe split this into `isConfigOption` that does the `ConfigOption` check and `shouldBeDocumented` that does the deprecation/exclusion checks.

Comment (tillrohrmann) [reply flink-docs/src/main/java/org/apache/flink/docs/configuration/ConfigOptionsDocGenerator.java]: Will split it up.

File: flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java (hunk 1 above)
Comment (tillrohrmann) [reply flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java]: Good idea. Will change it.

**Pull request merged**